# Advanced Statistics

# Assignment #3

# Part - II

Shamail Aamir Khan 

Eman Furrukh

# Q1) Part A:

Population of Interest: Customers of the mobile phone brand

Null Hypothesis: The number of satisfied customers is equal to or less than 52% (p <= 0.52)

Alternative Hypothesis: The number of satisfied customers is greater than 52% (p > 0.52)

# Q1) Part B:

In [1]:
import math
from statsmodels.stats.proportion import proportions_ztest

In [2]:
# Given information
n = 1000
p = 560 / n
H0 = 0.52

In [3]:
# Conduct z-test
count = p * n
stat, p_value = proportions_ztest(count, n, H0, alternative='larger')

In [4]:
# Print the results
print("Z-test statistic:", stat)
print("P-value:", p_value)

Z-test statistic: 2.54823595718813
P-value: 0.005413460638565616


# Q2) Part A:

Population of Interest: All member countries of the United Nations that participated in the survey

Null Hypothesis: The average GDP of the world is greater than or equal to 5400 (u >= 5400)

Alternative Hypothesis: The average GDP of the world is less than 5400 (u < 5400)

# Q2) Part B:

In [5]:
import pandas as pd
import statistics
from scipy import stats

In [6]:
# Load the dataset
df = pd.read_csv("Life_Expectancy_Data.csv")
# Print the first five rows
df.head()

Country  Year      Status  Life expectancy   Adult Mortality  \
0  Afghanistan  2015  Developing              65.0              263   
1  Afghanistan  2014  Developing              59.9              271   
2  Afghanistan  2013  Developing              59.9              268   
3  Afghanistan  2012  Developing              59.5              272   
4  Afghanistan  2011  Developing              59.2              275   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   ...  \
0             62     0.01               71.279624           65      1154  ...   
1             64     0.01               73.523582           62       492  ...   
2             66     0.01               73.219243           64       430  ...   
3             69     0.01               78.184215           67      2787  ...   
4             71     0.01                7.097109           68      3013  ...   

   Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0      6               8.16           65        0.1  584.259210  33700000.0   
1     58               8.18           62        0.1  612.696514    327582.0   
2     62               8.13           64        0.1  631.744976  31700000.0   
3     67               8.52           67        0.1  669.959000   3700000.0   
4     68               7.87           68        0.1   63.537231   2980000.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  

[5 rows x 22 columns]

In [7]:
# Compute the mean, standard deviation, and size of the "GDP" feature
mean = df["GDP"].mean()
std_dev = df["GDP"].std()
n = df["GDP"].size

print("\tGDP Stats")
print("Mean:", mean)
print("Standard Deviation:", std_dev)
print("Size:", n)

	GDP Stats
Mean: 5566.031886817592
Standard Deviation: 11475.90011727456
Size: 1649


In [8]:
pop_mean = 5400  
# Calculate Z-score
z_score = (mean - pop_mean) / (std_dev / (n ** 0.5))
# Calculate p-value
p_value = stats.norm.cdf(z_score)

In [9]:
print("Z-Score:", z_score)
print("p-Value:", p_value)

Z-Score: 0.5875097408137533
p-Value: 0.7215692948179755


# Q2) Part C:

The Null Hypothesis (H0) states that the average GDP of the world is greater than or equal to 5400. Whilst the alternative Hypothesis (H1) states that the average GDP of the world is less than 5400 (u < 5400)

Based on the Z-test results, we obtained a p-value of 0.722

Hence, we fail to reject H0 since value of p is greater than alpha i.e 0.05

# Q3) Part A:

Population of Interest: All the countries in the world

Step 2: Null Hypothesis: The life expectancy of the developed world is not significantly higher than the developing world

Alternative Hypothesis: The life expectancy of the developed world is significantly higher than the developing world

# Q3) Part B:

In [10]:
import pandas as pd
import statistics
from scipy import stats

In [11]:
# Load the dataset
df = pd.read_csv("Life_Expectancy_Data.csv")
# Print the first five rows
df.head()

Country  Year      Status  Life expectancy   Adult Mortality  \
0  Afghanistan  2015  Developing              65.0              263   
1  Afghanistan  2014  Developing              59.9              271   
2  Afghanistan  2013  Developing              59.9              268   
3  Afghanistan  2012  Developing              59.5              272   
4  Afghanistan  2011  Developing              59.2              275   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   ...  \
0             62     0.01               71.279624           65      1154  ...   
1             64     0.01               73.523582           62       492  ...   
2             66     0.01               73.219243           64       430  ...   
3             69     0.01               78.184215           67      2787  ...   
4             71     0.01                7.097109           68      3013  ...   

   Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0      6               8.16           65        0.1  584.259210  33700000.0   
1     58               8.18           62        0.1  612.696514    327582.0   
2     62               8.13           64        0.1  631.744976  31700000.0   
3     67               8.52           67        0.1  669.959000   3700000.0   
4     68               7.87           68        0.1   63.537231   2980000.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  

[5 rows x 22 columns]

In [12]:
# Create two lists for "Developing" and "Developed" countries
developing = df[df["Status"] == "Developing"]["Life expectancy "].tolist()
developed = df[df["Status"] == "Developed"]["Life expectancy "].tolist()

In [13]:
# Compute and print the mean, standard deviation, and size of "Life expectancy" in developing countries
developing_mean = statistics.mean(developing)
developing_std_dev = statistics.stdev(developing)
developing_size = len(developing)

print("\tStats for Developing Countries ")
print("Mean Life Expectancy:", developing_mean)
print("Standard Deviation of Life Expectancy:", developing_std_dev)
print("Size:", developing_size)

	Stats for Developing Countries 
Mean Life Expectancy: 67.68734896943852
Standard Deviation of Life Expectancy: 8.353982824185263
Size: 1407


In [14]:
# Compute and print the mean, standard deviation, and size of "Life expectancy" in developed countries
developed_mean = statistics.mean(developed)
developed_std_dev = statistics.stdev(developed)
developed_size = len(developed)

print("\tStats for Developed Countries ")
print("Mean Life Expectancy:", developed_mean)
print("Standard Deviation of Life Expectancy:", developed_std_dev)
print("Size:", developed_size)

	Stats for Developed Countries 
Mean Life Expectancy: 78.69173553719008
Standard Deviation of Life Expectancy: 4.273484527228104
Size: 242


In [16]:
# Perform Z-test for two populations
z_statistic, p_value = stats.ttest_ind(developing, developed)

In [17]:
# Print the results
print("Z-statistic:", z_statistic)
print("P-value:", p_value)

Z-statistic: -20.04208196882747
P-value: 3.9228534194275335e-80


# Q3) Part C:

The null hypothesis (H0) states that there is no significant difference in the life expectancy between the developed and developing world. The alternative hypothesis (H1) states that the life expectancy of the developed world is significantly higher than the developing world.

Based on the Z-test results, we obtained a p-value of 3.93e^-80

Hence, we have rejected the null hypothesis